File: src/main/java/com/blog_app_apis/service/CategoryService.java
package com.blog_app_apis.service;

import com.blog_app_apis.dtos.CategroyDTO;

import java.util.List;

public interface CategoryService {
    // create
    CategroyDTO createCategory(CategroyDTO categroyDTO);

    // update
    CategroyDTO updateCategory(CategroyDTO categroyDTO, Integer categoryId);

    //delete
    void deleteCategory(Integer categoryId);

    // get
    CategroyDTO getCategory(Integer categoryId);

    // getAll
    List<CategroyDTO> getCategories();
}

File: blog-app-react/src/services/categoryService.js
import api from './api';

export const getCategories = async () => {
  const response = await api.get('/api/category');
  return response.data;
};

export const getCategoryById = async (categoryId) => {
  const response = await api.get(`/api/category/${categoryId}`);
  return response.data;
};

export const createCategory = async (categoryData) => {
  const response = await api.post('/api/category', categoryData);
  return response.data;
};

export const updateCategory = async (categoryData, categoryId) => {
  const response = await api.put(`/api/category/${categoryId}`, categoryData);
  return response.data;
};

export const deleteCategory = async (categoryId) => {
  const response = await api.delete(`/api/category/${categoryId}`);
  return response.data;
};
